src/middlewares/request-unflattener.js

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash';
```

Comment:
A lot of this are going on there, is there any reason why you did not make a true class for this ?

Reply:
Maybe we could have put this code in the FieldsFlattener, and create a function which returns a middleware 🤷‍♂️

Reply:
Okay, we can create a `Flattener` service, let's do it from the main branches once we put our work together